File: component-acgnews/src/main/java/com/rabtman/acgnews/mvp/model/jsoup/ZeroFiveNewsPage.java
package com.rabtman.acgnews.mvp.model.jsoup;

import com.fcannizzaro.jsoup.annotations.interfaces.Attr;
import com.fcannizzaro.jsoup.annotations.interfaces.Items;
import com.fcannizzaro.jsoup.annotations.interfaces.Selector;
import java.util.List;

/**
 * @author Rabtman
 */
@Selector("div.w article-list-page clearfix")
public class ZeroFiveNewsPage {

  @Items
  private List<ZeroFiveNews> zeroFiveNewsList;
  @Attr(query = "div div.pages div ul li a:last-child", attr = "href")
  private String pageCount;

  public List<ZeroFiveNews> getZeroFiveNewsList() {
    return zeroFiveNewsList;
  }

  public void setZeroFiveNewsList(List<ZeroFiveNews> zeroFiveNewsList) {
    this.zeroFiveNewsList = zeroFiveNewsList;
  }

  public String getPageCount() {
    return pageCount.substring(pageCount.lastIndexOf("_") + 1, pageCount.lastIndexOf("."));
  }

  public void setPageCount(String pageCount) {
    this.pageCount = pageCount;
  }

  @Override
  public String toString() {
    return "ZeroFiveNewsPage{" +
        "zeroFiveNewsList=" + zeroFiveNewsList +
        ", pageCount=" + pageCount +
        '}';
  }
}

File: component-acgnews/src/main/java/com/rabtman/acgnews/mvp/model/jsoup/ZeroFiveNewsPage.kt
package com.rabtman.acgnews.mvp.model.jsoup

import com.fcannizzaro.jsoup.annotations.interfaces.Attr
import com.fcannizzaro.jsoup.annotations.interfaces.Items
import com.fcannizzaro.jsoup.annotations.interfaces.Selector

/**
 * @author Rabtman
 */
@Selector("div.w article-list-page clearfix")
class ZeroFiveNewsPage {
    @Items
    var zeroFiveNewsList: List<ZeroFiveNews>? = null

    @Attr(query = "div div.pages div ul li a:last-child", attr = "href")
    private var pageCount: String? = null

    fun getPageCount(): Int {
        pageCount?.let {
            it.substring(it.lastIndexOf("_") + 1, it.lastIndexOf(".")).toIntOrNull()?.let { count ->
                return count
            } ?: run {
                return 1
            }
        } ?: run {
            return 1
        }
    }

    fun setPageCount(pageCount: String?) {
        this.pageCount = pageCount
    }

    override fun toString(): String {
        return "ZeroFiveNewsPage{" +
                "zeroFiveNewsList=" + zeroFiveNewsList +
                ", pageCount=" + pageCount +
                '}'
    }
}